src/store/modules/map/index.js

```diff
@@ -0,0 +1,24 @@
+import buildWmsLayer from '~/lib/build-wms-layer'
+
+export default {
+  namespaced: true,
+
+  state: () => ({
+    layers: [],
+  }),
+
+  getters: {},
+
+  mutations: {
+    SET_ACTIVE_LAYERS(state, { layers }) {
+      const wmsLayers = layers.map(layer => buildWmsLayer(layer))
+      state.layers = wmsLayers
+    },
```

Comment:
Currently we rebuild all the `wmsLayers` every time, and re-render them completely on the map. This can be done more efficiently, which also gives a much nicer effect for the user (just drawing the newly selected layer on top of the already selected layers). You can diff the existing `wmsLayers` against the passed selected `layers` and only add/remove the layers that are different.